docs/02-app/01-building-your-application/04-caching/02-router-cache.mdx

```diff
@@ -0,0 +1,78 @@
+
+## How it works
+
+When the user first visits a route, Next.js will cache the rendered result of the route segments (layouts and page). If the route contains `<Link>` components (or uses `router.prefetch`), Next.js will automatically start prefetching and caching the content of those linked routes.
+
+As users navigate between routes, Next.js will check if the route segments are already cached. If so, it will skip sending a new request to the server for rendering, and use the payload of cached segments instead. This works when navigating to new routes, as well as backwards and forwards browser navigation.
+
+> **What is the rendered result?**
+>
+> You may see the terms like **RSC Payload**, **Rendered Result**, and **Flight Data** used interchangeably. They all refer to the same thing: the rendered result of Server Components, which is sent to the client. To learn more, see the [Server Components]() documentation.
+
+## Duration and Invalidation
+
+The cache is stored in the browser's temporary memory. There are two factors that determine how long the router cache lasts:
+
+- **Session**: The cache persists across soft navigation. However, it's cleared on page refresh or [hard navigation]().
+- **Automatic invalidation period**: The cache of an individual segment is automatically invalidated after a time period. The duration depends on whether the route is [static]() or [dynamic]():
+  - **Dynamic routes**: 30 seconds
+  - **Static routes**: 5 minutes
+
+While a page refresh or hard navigation will clear **all** cached segments, the automatic invalidation period only affects the individual segment from the time it was last accessed or created.
+
+## APIs
+
+The following APIs affect the Router Cache:
+
+### `<Link>`
+
+By default, the `<Link>` component automatically prefetches linked routes. This adds the rendered result to the cache.
+
+To disable prefetching, you can set the `prefetch` prop to `false`.
+
+Learn more about the [`<Link>` component]().
+
+### `router.prefetch`
+
+The `prefetch` option of the `useRouter` hook can be used to manually prefetch a route. This adds the rendered result to the cache.
+
+See the [`useRouter` hook]() API reference.
+
+### `router.refresh`
+
+The `refresh` method of the `useRouter` hook can be used to manually refresh a route. This completely clears the router cache, makes a new request to the server, refetches data (which may be cached - see [Interactions]()), and re-renders Server Components from the root layout down to the page.
+
+The rendered result will be merged with client components without losing the client-side React state or browser state (e.g. scroll position).
+
+See the [`useRouter` hook]() API reference.
+
+### `revalidatePath` and `revalidateTag`
+
+The `revalidatePath` and `revalidateTag` methods invalidates the [data cache]() below the path provided, which in turn invalidates the Router Cache.
```

Comment:
The router cache is invalidated for only a client that calls `revalidatePath` or `revalidateTag` through a Server Actions, isn't it?
It would be appreciated to clarify whose router cache will be invalidated.

Reply:
Yes, it is. I've moved the note to caching page now.

Reply:
Thanks for the update! But I would like to clarify something more.

Please imagine that user A and B are viewing the same page at the same time. If user A invokes Server Actions and it calls `revalidatePath`/`Tag`, it will invalidate the Data cache, the Render cache, and the only user A's Router cache.

However, user B's Router cache would not be invalidated, so if the page was statically rendered and prefetched, it would not be up-to-date for 5 minutes. This may seem obvious, but I don't think this is well understood by App Router developers at this time.